lib/slack/real_time/concurrency/eventmachine.rb

```diff
@@ -30,7 +30,23 @@ def ensure_reactor_running
           end
 
           def connect
-            @driver = ::Faye::WebSocket::Client.new(url, nil, options)
```

Comment:
Did you try inheriting from `Faye::WebSocket::Client` and overriding `parse` and `write` methods? And instantiate the child class here?